### PR TITLE
chore: loading data right from ChatMessage #254

### PR DESCRIPTION
**Pull request by FrolYo**
Refactored chat data load logic so that every ChatMessage loads required data on demand instead of loading all required data for whole list.

### CODE REVIEW

File: src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/ChatMessage.razor [hunk 1]
```diff
@@ -99,48 +104,79 @@
...
-        }
         await TimeZoneConverter.WhenInitialized.ConfigureAwait(false);
     }
 
-    public override Task SetParametersAsync(ParameterView parameters)
-        => this.HasChangedParameters(parameters) ? base.SetParametersAsync(parameters) : Task.CompletedTask;
+    protected override async Task<Model?> ComputeState(CancellationToken cancellationToken) {
+        var principalId = await ChatAuthor
+            .GetOwnPrincipalId(Session, Message.Entry.ChatId, CancellationToken.None)
+            .ConfigureAwait(true);
```
Comment (evgeniy-ac) [inline src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/ChatMessage.razor:122]: Other are `.ConfigureAwait(false)`

File: src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/ChatMessage.razor [hunk 2]
```diff
@@ -1,93 +1,98 @@
...
 @using Microsoft.AspNetCore.Components.Rendering
 @using System.Diagnostics.CodeAnalysis
 @{
-    var entry = Model.Entry;
-    var markup = Model.Markup;
-    var attachments = Model.Attachments;
+    var model = State.LatestNonErrorValue;
+}
+
+@if (model != null) {
```
[the remark this answers is not in the document]
Comment (FrolYo) [reply src/dotnet/Chat.UI.Blazor/Components/ChatView/Items/ChatMessage.razor]: done

File: src/dotnet/Chat.UI.Blazor/Services/ChatEntryReader.cs [hunk 3]
```diff
@@ -25,6 +25,15 @@ public ChatEntryReader(IChats chats, Session session, Symbol chatId, ChatEntryTy
         return tile.Entries.SingleOrDefault(e => e.Id == id);
     }
 
+    public async Task<TextEntryAttachment> GetFirstAttachment(long id, CancellationToken cancellationToken)
+    {
+        var all = await Chats.GetTextEntryAttachments(Session, ChatId, id, cancellationToken).ConfigureAwait(false);
+        return all.First();
+    }
+
+    public Task<ImmutableArray<TextEntryAttachment>> GetAttachments(long id, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (FrolYo) [reply src/dotnet/Chat.UI.Blazor/Services/ChatEntryReader.cs:34]: done

File: src/dotnet/Chat.UI.Blazor/Services/ChatEntryReader.cs [hunk 4]
```diff
@@ -25,6 +25,15 @@ public ChatEntryReader(IChats chats, Session session, Symbol chatId, ChatEntryTy
         return tile.Entries.SingleOrDefault(e => e.Id == id);
     }
 
+    public async Task<TextEntryAttachment> GetFirstAttachment(long id, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (FrolYo) [reply src/dotnet/Chat.UI.Blazor/Services/ChatEntryReader.cs:28]: done

**Review by alexis-kochetov — APPROVED** [review]

**Pull request merged**